src/transformers/modeling_fx_utils.py

```diff
@@ -0,0 +1,226 @@
+import dis
+import inspect
+from typing import List, Optional, Union
+
+import torch
+from torch.fx import GraphModule, Node, Proxy, Tracer
+
+from . import PreTrainedModel
+
+
+class CustomProxy(Proxy):
+    def __init__(self, node: Node, tracer: Optional[Tracer] = None):
```

Comment:
A small docstring on what that object does would go a long way for the code maintainability in the future :-)
What does this custom proxy does that the torch.fx does not?

Also should if be `HFProxy` instead of `CustomProxy`?